third_party/gems/test/smoke_test.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+# This tests that the bundler wrapper produced by the `@gems` package is able to
+# install a gem, and provide an environment with that gem in scope.
+
+# --- begin runfiles.bash initialization ---
+# Copy-pasted from Bazel's Bash runfiles library https://github.com/bazelbuild/bazel/blob/defd737761be2b154908646121de47c30434ed51/tools/bash/runfiles/runfiles.bash
+set -euo pipefail
+if [[ ! -d "${RUNFILES_DIR:-/dev/null}" && ! -f "${RUNFILES_MANIFEST_FILE:-/dev/null}" ]]; then
+  if [[ -f "$0.runfiles_manifest" ]]; then
+    export RUNFILES_MANIFEST_FILE="$0.runfiles_manifest"
+  elif [[ -f "$0.runfiles/MANIFEST" ]]; then
+    export RUNFILES_MANIFEST_FILE="$0.runfiles/MANIFEST"
+  elif [[ -f "$0.runfiles/bazel_tools/tools/bash/runfiles/runfiles.bash" ]]; then
+    export RUNFILES_DIR="$0.runfiles"
+  fi
+fi
+if [[ -f "${RUNFILES_DIR:-/dev/null}/bazel_tools/tools/bash/runfiles/runfiles.bash" ]]; then
+  # shellcheck disable=SC1090
+  source "${RUNFILES_DIR}/bazel_tools/tools/bash/runfiles/runfiles.bash"
+elif [[ -f "${RUNFILES_MANIFEST_FILE:-/dev/null}" ]]; then
+  # shellcheck disable=SC1090
+  source "$(grep -m1 "^bazel_tools/tools/bash/runfiles/runfiles.bash " \
+            "$RUNFILES_MANIFEST_FILE" | cut -d ' ' -f 2-)"
+else
+  echo >&2 "ERROR: cannot find @bazel_tools//tools/bash/runfiles:runfiles.bash"
+  exit 1
+fi
+# --- end runfiles.bash initialization ---
+
+# put bundle in the path
+# shellcheck disable=SC1083
+PATH="$(dirname "$(rlocation %{workspace}/bundler/bundle)"):$PATH"
+export PATH
+
+# bundle requires $HOME
+HOME=$PWD
+export HOME
+
+# ensure that bundle works
+bundle --help
+
+
+# shellcheck disable=SC1083
```

Comment:
I think we should disable this error code. Can you audit this PR to find instances where we can replace

```bash
cp "$(rlocation %{workspace}/test/Gemfile)" .
```

with

```bash
cp "$(rlocation "%{workspace}/test/Gemfile")" .
```

? The error it's saying is that "you might have meant to do `echo {1,2}`" which has a special meaning in the shell (expands to `echo 1 2`). We should be able to say we don't care about shell expansion.

Reply:
^ edited (never* disable)

Reply:
Ah, sorry. I'll go through and remove those.

Reply:
I've removed all instances of SC1083 being disabled.